cloud/deployment/deployment.go

```diff
@@ -274,103 +288,274 @@ func Create(label, workspaceID, description, clusterID, runtimeVersion, dagDeplo
 	}
 
 	// select and validate cluster
-	clusterID, err = useSharedClusterOrSelectDedicatedCluster(cloudProvider, region, c.Organization, clusterID, clusterType, coreClient)
-	if err != nil {
-		return err
-	}
-
-	scheduler := astro.Scheduler{
-		AU:       schedulerAU,
-		Replicas: schedulerReplicas,
-	}
-
-	spec := astro.DeploymentCreateSpec{
-		Executor:  executor,
-		Scheduler: scheduler,
+	if !IsDeploymentStandard(deploymentType) {
+		clusterID, err = selectCluster(clusterID, c.Organization, platformCoreClient)
+		if err != nil {
+			return err
+		}
 	}
 
-	if dagDeploy == "enable" { //nolint: goconst
+	if dagDeploy == enable { //nolint: goconst
 		dagDeployEnabled = true
 	} else {
 		dagDeployEnabled = false
 	}
+	createDeploymentRequest := astroplatformcore.CreateDeploymentRequest{}
+	if IsDeploymentStandard(deploymentType) || IsDeploymentDedicated(deploymentType) {
+		var highAvailabilityValue bool
+		if highAvailability == enable { //nolint: goconst
+			highAvailabilityValue = true
+		} else {
+			highAvailabilityValue = false
+		}
+		var workerConcurrency int
+		for i := range configOption.WorkerMachines {
+			if strings.EqualFold(configOption.DefaultValues.WorkerMachineName, configOption.WorkerMachines[i].Name) {
+				workerConcurrency = int(configOption.WorkerMachines[i].Concurrency.Default)
+			}
+		}
+		defautWorkerQueue := []astroplatformcore.WorkerQueueRequest{{
+			AstroMachine:      astroplatformcore.WorkerQueueRequestAstroMachine(configOption.DefaultValues.WorkerMachineName),
+			IsDefault:         true,
+			MaxWorkerCount:    int(configOption.WorkerQueues.MaxWorkers.Default),
+			MinWorkerCount:    int(configOption.WorkerQueues.MinWorkers.Default),
+			Name:              "default",
+			WorkerConcurrency: workerConcurrency,
+		}}
+		if defaultTaskPodCpu == "" {
+			defaultTaskPodCpu = configOption.ResourceQuotas.DefaultPodSize.Cpu.Default
+		}
+		if defaultTaskPodMemory == "" {
+			defaultTaskPodMemory = configOption.ResourceQuotas.DefaultPodSize.Memory.Default
+		}
+		if resourceQuotaCpu == "" {
+			resourceQuotaCpu = configOption.ResourceQuotas.ResourceQuota.Cpu.Default
+		}
+		if resourceQuotaMemory == "" {
+			resourceQuotaMemory = configOption.ResourceQuotas.ResourceQuota.Memory.Default
+		}
+		// build standard input
+		if IsDeploymentStandard(deploymentType) {
+			var requestedCloudProvider astroplatformcore.CreateStandardDeploymentRequestCloudProvider
+			switch cloudProvider {
+			case gcpCloud:
+				requestedCloudProvider = astroplatformcore.CreateStandardDeploymentRequestCloudProviderGCP
+			case awsCloud:
+				requestedCloudProvider = astroplatformcore.CreateStandardDeploymentRequestCloudProviderAWS
+			case azureCloud:
+				requestedCloudProvider = astroplatformcore.CreateStandardDeploymentRequestCloudProviderAZURE
+			default:
+				// Handle the case when cloudProvider doesn't match any of the known values
+				fmt.Println("Unknown cloud provider:", cloudProvider)
```

Comment:
should we err here instead right away

Reply:
is it even possible to get here? if so we can add a comment

Reply:
I don't think it's possible to get here. Just going to add a comment